### PR TITLE
Fix static analyzer possible leak

### PR DESCRIPTION
**Pull request by julek-wolfssl**
crl would never be null there but clean up code to make sure newcrl->crlLock gets free'd

### CODE REVIEW

**Comment by julek-wolfssl** [comment]
Retest this please

File: src/crl.c [hunk 1]
```diff
@@ -837,20 +838,19 @@ int wolfSSL_X509_STORE_add_crl(WOLFSSL_X509_STORE *store, WOLFSSL_X509_CRL *newc
...
         }
-        if (DupX509_CRL(crl, newcrl) != 0) {
-            if (crl != NULL) {
-                wc_UnLockRwLock(&newcrl->crlLock);
-                FreeCRL(crl, 1);
-            }
+        ret = DupX509_CRL(crl, newcrl);
+        wc_UnLockRwLock(&newcrl->crlLock);
+        if (ret != 0) {
+            FreeCRL(crl, 1);
```
[the remark this answers is not in the document]
Comment (bandi13) [reply src/crl.c:851]: `FreeCRL` should also check `crl != NULL` and return BAD_ARG_ERR or something. It will crash if NULL is passed.

**Review by bandi13 — CHANGES_REQUESTED** [review]
I have some questions that should be addressed.

File: src/crl.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply src/crl.c:851]: `crl` will never be `NULL` here as its checked right after allocation. I agree that `FreeCRL` should be performing a check internally though.

**Review by bandi13 — APPROVED** [review]

**Pull request merged**